br/pkg/config/kv.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 PingCAP, Inc. Licensed under Apache-2.0.
+package config
+
+import (
+	"encoding/json"
+
+	"github.com/docker/go-units"
+)
+
+type ConfigSet[T uint | uint64] struct {
+	Value  T
+	HasSet bool
+}
+
+type KVConfig struct {
+	ImportGoroutines    ConfigSet[uint]
+	MergeRegionSize     ConfigSet[uint64]
+	MergeRegionKeyCount ConfigSet[uint64]
+}
+
+func ParseImportThreadsFromConfig(resp []byte) (uint, error) {
+	type importer struct {
+		Threads uint `json:"num-threads"`
+	}
+
+	type config struct {
+		Import importer `json:"import"`
+	}
+	var c config
+	e := json.Unmarshal(resp, &c)
+	if e != nil {
+		return 0, e
+	}
+
+	return c.Import.Threads, nil
+}
+
+func ParseMergeRegionSizeFromConfig(resp []byte) (uint64, uint64, error) {
```

Comment:
It seems this function is only called in `conn` package and doesn't rely on any item in `config`. Why not make it a private function in `conn`?

Reply:
I'd like to put all parse config logic into one pkg for future usage.